tests/pipelines/test_base_pipeline.py

```diff
@@ -43,6 +43,51 @@ def some_pipeline(step_):
     return pipeline_instance
 
 
+def test_initialize_pipeline():
```

Comment:
Can we break these down into different tests, rather than one big one?

Reply:
Yeah that would be much nicer, I just didn't do it because of the same @pipeline/@step declaration, I'll see if I can create a fixture for that